src/otx/cli/utils/installation.py

```diff
@@ -0,0 +1,530 @@
+from importlib.util import find_spec
+from pathlib import Path
+from warnings import warn
+
+import pkg_resources
+from pkg_resources import Requirement
+
+AVAILABLE_TORCH_VERSIONS = {
+    "1.13.0": {"torchvision": "0.14.0", "cuda": ("11.6", "11.7")},
+    "1.13.1": {"torchvision": "0.14.1", "cuda": ("11.6", "11.7")},
+    "2.0.0": {"torchvision": "0.15.1", "cuda": ("11.7", "11.8")},
+    "2.0.1": {"torchvision": "0.15.2", "cuda": ("11.7", "11.8")},
+    "2.1.1": {"torchvision": "0.16.1", "cuda": ("11.8", "12.1")},
+}
+
+MM_REQUIREMENTS = [
+    "mmcv",
+    "mmcv-full",
+    "mmengine",
+    "mmdet",
+    "mmsegmentation",
+    "mmpretrain",
+    "mmdeploy",
+    "mmaction2",
+]
+
+
+def get_requirements(module: str = "otx") -> dict[str, list[Requirement]]:
+    """Get requirements of module from importlib.metadata.
+
+    This function returns list of required packages from importlib_metadata.
+
+    Example:
+        >>> get_requirements("otx")
+        {
+            "api": ["attrs>=21.2.0", ...],
+            "anomaly": ["anomalib==0.5.1", ...],
+            ...
+        }
+
+    Returns:
+        Dict[str, List[Requirement]]: List of required packages for each optional-extras.
+    """
+    requirement_list: list[str] | None = requires(module)
+    extra_requirement: dict[str, list[Requirement]] = {}
+    if requirement_list is None:
+        return extra_requirement
+    for requirement in requirement_list:
+        extra = "api"
+        requirement_extra: list[str] = requirement.replace(" ", "").split(";")
+        if isinstance(requirement_extra, list) and len(requirement_extra) > 1:
+            extra = requirement_extra[-1].split("==")[-1].strip("'\"")
+        _requirement_name = requirement_extra[0]
+        _requirement = Requirement.parse(_requirement_name)
+        if extra in extra_requirement:
+            extra_requirement[extra].append(_requirement)
+        else:
+            extra_requirement[extra] = [_requirement]
+    return extra_requirement
+
+
+def parse_requirements(
+    requirements: list[Requirement],
+) -> tuple[str, list[str], list[str]]:
+    """Parse requirements and returns torch, mmcv and task requirements.
+
+    Args:
+        requirements (list[Requirement]): List of requirements.
+
+    Raises:
+        ValueError: If torch requirement is not found.
+
+    Examples:
+        >>> requirements = [
+        ...     Requirement.parse("torch==1.13.0"),
+        ...     Requirement.parse("mmcv-full==1.7.0"),
+        ...     Requirement.parse("mmcls==0.12.0"),
+        ...     Requirement.parse("onnx>=1.8.1"),
+        ... ]
+        >>> parse_requirements(requirements=requirements)
+        (Requirement.parse("torch==1.13.0"),
+        [Requirement.parse("mmcv-full==1.7.0"), Requirement.parse("mmcls==0.12.0")],
+        Requirement.parse("onnx>=1.8.1"))
+
+    Returns:
+        tuple[Requirement, list[Requirement], list[Requirement]]: Tuple of torch, mmcv and other requirements.
```

Comment:
This function returns `tuple[str, list[str], list[str]]`. But, why is this docstring not aligned with the code.

Reply:
Thank you. I must have missed something, I went back through the docsting and made the necessary changes.
- https://github.com/openvinotoolkit/training_extensions/pull/2679/commits/39c65454f5bd5aeff26fbb60925e0f1064b993bb